src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -864,6 +885,29 @@ def pin_subvolume_group(self, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    def subvolume_group_exists(self, **kwargs):
+        volname = kwargs['vol_name']
+        ret = 0, "", ""
+        volume_exists = False
+
+        try:
+            with open_volume(self, volname) as fs_handle:
+                volume_exists = True
+                try:
+                    fs_handle.opendir(self.volspec.base_dir)
```

Comment:
Instead of this, could you please check does the ENOENT traceback contains the path. If so, you can rely on that and get away with `volume_exists` flag and this `opendir` as well.

Same needs to be done for `subvolume_exists` as well.